packages/api-client/src/tcp/ReconnectingWebsocket.ts

```diff
@@ -82,6 +83,22 @@ export class ReconnectingWebsocket {
     }
 
     this.hasUnansweredPing = false;
+
+    /**
+     * According to https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine, navigator.onLine attribute and 'online' and 'offline' events are not reliable enough (especially when it's truthy).
+     * We won't receive the 'offline' event when the system goes to sleep (e.g. closing the lid of a laptop).
+     * In this case navigator.onLine will still return true, but the WebSocket connection could be closed.
+     * To handle this, we need a custom approach to detect when the system goes to sleep and when it wakes up.
+     * **/
+    onBackFromSleep({
+      callback: () => {
+        if (this.socket) {
+          this.logger.debug('Back from sleep, reconnecting WebSocket');
+          this.socket?.reconnect();
+        }
+      },
+      isDisconnected: () => this.getState() === WEBSOCKET_STATE.CLOSED,
```

Comment:
Wasn't the ping/pong mechanism enough to detect that the connection was closed? 
Is there some kind of timer that runs to check if a message has been received in a certain amount of time?

Reply:
Yeah, we have the mechanism for that, but it's a ping every 20 seconds, which doesn't seem to be efficient enough to reconnect immediately after coming back from sleep. From what I was testing, it was the 20s that we had to wait for the weboscket to reconnect after coming back from sleep.

Reply:
Humm that makes sense, 20s is, indeed, pretty long time. 

Then your solution seems solid to me 😎